### PR TITLE
Adding debug logs and fixing some bugs in AzureRG task.

### PR DESCRIPTION
**Pull request by aydey**

### CODE REVIEW

File: Tasks/AzureResourceGroupDeployment/operations/AzureUtil.ts [hunk 1]
```diff
@@ -64,35 +64,43 @@ export class AzureUtil {
...
                 continue;
             }
-
+            tl.debug("Mapping for load balancer: " + lb.name);
             var loadBalancer = new LoadBalancer();
             var publicAddressId = lb.properties.frontendIPConfigurations[0].properties.publicIPAddress.id;
             loadBalancer.FrontEndPublicAddress = fqdns[publicAddressId];
             loadBalancer.Id = lb.id;
 
+            tl.debug("Frontend IP address is: " + JSON.stringify(loadBalancer.FrontEndPublicAddress));
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/AzureUtil.ts]: do we need this, you have already printed the whole thing below

Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/AzureUtil.ts]: do we need this, you have already printed the whole thing below

File: Tasks/Common/azure-arm-rest/AzureServiceClient.ts [hunk 2]
```diff
@@ -170,6 +170,11 @@ export class ServiceClient {
             nextRequest.method = 'GET';
             nextRequest.uri = nextLinkUrl;
             var response = await this.beginRequest(nextRequest);
+            if (tl.getVariable('system.debug.extended') && tl.getVariable('system.debug.extended').toLowerCase() == 'true') {
```
Comment (sachinma) [inline Tasks/Common/azure-arm-rest/AzureServiceClient.ts]: remove it from all places and move it to webClient.ts if required

Comment (ds-ms) [inline Tasks/Common/azure-arm-rest/AzureServiceClient.ts]: +1

**Comment by ds-ms** [comment]
You should probably bump up the version of other tasks which consume this lib for updating it in the next release.

**Pull request merged**